src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -143,30 +165,33 @@ struct AreaStructureWrapper : public chip::app::Clusters::ServiceArea::Structs::
             areaDesc.locationInfo.SetNull();
         }
 
-        areaDesc.landmarkTag = aLandmarkTag;
-        areaDesc.positionTag = aPositionTag;
-        areaDesc.surfaceTag  = aSurfaceTag;
-
-        // this assumes areaDesc structure was created above, if appropriate
-        if (!areaDesc.locationInfo.IsNull())
+        // todo improve this when addressing issue https://github.com/project-chip/connectedhomeip/issues/34519
+        if (aLandmarkTag.IsNull())
         {
-            if (aLocationName.empty())
-            {
-                areaDesc.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, 0);
-            }
-            else if (aLocationName.size() > sizeof(mLocationNameBuffer))
+            areaDesc.landmarkInfo.SetNull();
+        }
+        else
+        {
+            areaDesc.landmarkInfo.SetNonNull();
+            areaDesc.landmarkInfo.Value().landmarkTag = aLandmarkTag.Value();
+            if (aRelativePositionTag.IsNull())
             {
-                // Save the truncated name that fits into available size.
-                memcpy(mLocationNameBuffer, aLocationName.data(), sizeof(mLocationNameBuffer));
-                areaDesc.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, sizeof(mLocationNameBuffer));
+                areaDesc.landmarkInfo.Value().positionTag.SetNull();
             }
             else
             {
-                // Save full name.
-                memcpy(mLocationNameBuffer, aLocationName.data(), aLocationName.size());
-                areaDesc.locationInfo.Value().locationName = CharSpan(mLocationNameBuffer, aLocationName.size());
+                areaDesc.landmarkInfo.Value().positionTag.SetNonNull();
+                areaDesc.landmarkInfo.Value().positionTag.Value() = aRelativePositionTag.Value();
             }
         }
+
+        // this assumes areaDesc structure was created above, if appropriate
+        if (!areaDesc.locationInfo.IsNull())
+        {
+            auto sizeToCopy = std::min(sizeof(mLocationNameBuffer), aLocationName.size());
```

Comment:
same - use char span functions to do a copy here.

Reply:
@cecille Same as above - tracked using https://github.com/project-chip/connectedhomeip/issues/34858. We'll address it in a subsequent PR